core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -16,158 +16,169 @@
+  'skill_description': string;
+}
+
+export interface FetchSuggestionsResponse {
+  'target_id_to_opportunity_dict': {
+    [targetId: string]: OpportunityDict;
+  };
+  suggestions: SuggestionBackendDict[];
+}
+
+interface FetchSuggestions {
+  [targetId: string]: {
+    suggestion: SuggestionBackendDict;
+    details: OpportunityDict;
+  };
+}
 
-    var _fetchSuggestionsAsync = async function(url) {
-      return $http.get(url).then(function(res) {
-        var suggestionIdToSuggestions = {};
-        var targetIdToDetails = res.data.target_id_to_opportunity_dict;
-        res.data.suggestions.forEach(function(suggestion) {
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewService {
+  constructor(
+    private contributionAndReviewBackendApiService:
+      ContributionAndReviewBackendApiService
+  ) {}
+
+  private async _fetchSuggestionsAsync(
+      fetchType: string
+  ): Promise<FetchSuggestions> {
+    return (
+      this.contributionAndReviewBackendApiService.fetchSuggestionsAsync(
+        fetchType
+      ).then((responseBody) => {
+        const suggestionIdToSuggestions: FetchSuggestions = {};
+        const targetIdToDetails = responseBody.target_id_to_opportunity_dict;
+        responseBody.suggestions.forEach((suggestion) => {
           suggestionIdToSuggestions[suggestion.suggestion_id] = {
             suggestion: suggestion,
             details: targetIdToDetails[suggestion.target_id]
           };
         });
         return suggestionIdToSuggestions;
+      })
+    );
+  }
+
+  async getUserCreatedQuestionSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('SUBMITTED_QUESTION_SUGGESTIONS');
+  }
+
+  async getReviewableQuestionSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('REVIEWABLE_QUESTION_SUGGESTIONS');
+  }
+
+  async getUserCreatedTranslationSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('SUBMITTED_TRANSLATION_SUGGESTIONS');
+  }
+
+  async getReviewableTranslationSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('REVIEWABLE_TRANSLATION_SUGGESTIONS');
+  }
+
+  resolveSuggestionToExploration(
+      targetId: string, suggestionId: string, action: string,
+      reviewMessage: string, commitMessage: string,
+      onSuccess: (suggestionId: string) => void,
+      onFailure: (error) => void
+  ): Promise<void> {
+    const requestBody = {
+      action: action,
+      review_message: reviewMessage,
+      commit_message: (
+        action === AppConstants.ACTION_ACCEPT_SUGGESTION ?
+        commitMessage : null
+      )
+    };
+
+    return this.contributionAndReviewBackendApiService
+      .reviewExplorationSuggestionAsync(
+        targetId, suggestionId, requestBody
+      ).then(() => {
+        onSuccess(suggestionId);
+      }, (error) => {
+        onFailure && onFailure(error);
       });
+  }
+
+  resolveSuggestiontoSkill(
```

Comment:
Need to update function name to reflect naming changes.

Reply:
done